scheme-service/src/main/java/uk/gov/justice/laa/fee/scheme/feecalculators/CalculateMediationFee.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.justice.laa.fee.scheme.feecalculators;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import uk.gov.justice.laa.fee.scheme.entity.FeeEntity;
+import uk.gov.justice.laa.fee.scheme.model.FeeCalculation;
+import uk.gov.justice.laa.fee.scheme.model.FeeCalculationRequest;
+import uk.gov.justice.laa.fee.scheme.model.FeeCalculationResponse;
+
+/**
+ * Calculate the mediation fee for a given fee entity and fee data.
+ * */
+public final class CalculateMediationFee {
+
+  /**
+   * Determines whether the calculation should include mediation sessions based presence of numberOfMediationSessions.
+   */
+  public static FeeCalculationResponse getFee(FeeEntity feeEntity, FeeCalculationRequest feeData) {
+    String feeCode = feeData.getFeeCode();
+    Integer numberOfMediationSessions = feeData.getNumberOfMediationSessions();
+
+    if (numberOfMediationSessions == null) {
+      return getCalculationWithoutMediationSessions(feeCode, feeEntity, feeData);
+    } else {
+      return getCalculationWithMediationSessions(feeCode, feeEntity, feeData);
+    }
+  }
+
+  /**
+   * Gets fixed fee depending on number if mediation sessions.
+   */
+  private static FeeCalculationResponse getCalculationWithMediationSessions(String feeCode, FeeEntity feeEntity,
+                                                                            FeeCalculationRequest feeData) {
```

Comment:
feeCode is available in FeeCalculationRequest, so
```suggestion
  private static FeeCalculationResponse getCalculationWithMediationSessions(FeeEntity feeEntity,
                                                                            FeeCalculationRequest feeData) {
```